File: 125_ValidPalindrome.py
import re

class Solution(object):
    def isPalindrome(self, s):
        s = re.sub('[^A-Za-z0-9]+', '', s)
        s = s.lower()
        stringLength = len(s)
        
        if(stringLength <= 1):
            return True
        for i,el in enumerate(s):
            if(el != s[stringLength-i-1]):
                return False
        return True

File: 125_ValidPalindrome.cpp
class Solution {
public:
    string newStr(string s) {
        string newStr = "";

        for(int i = 0; i < s.size(); i++) {
            char currentLetter = s[i];

            if(isalpha(currentLetter) || isdigit(currentLetter)) {
                newStr += tolower(currentLetter);
            }
        }
        return newStr;
    }
    bool isPalindrome(string s) {
        s = newStr(s);
        int strLen = s.size();

        for(int i = 0; i < strLen; i++) {
            char currentLetter = s[i];
            char oppositeLetter = s[strLen - i - 1];
            
            if(currentLetter != oppositeLetter) {
                return false;
            }
        }
        return true;
    }
};
